src/plugins/graph/adornments/movable-line/movable-line.tsx

```diff
@@ -86,198 +96,246 @@ export const MovableLine = observer(function MovableLine(props: IProps) {
+      lineObject.cover && fixEndPoints(lineObject.cover, pixelPtsOnAxes);
+      lineObject.arrowLower && fixArrow(lineObject.arrowLower, 1, pixelPtsOnAxes);
+      lineObject.arrowUpper && fixArrow(lineObject.arrowUpper, 2, pixelPtsOnAxes);
+      lineObject.handleLower && fixHandles(lineObject.handleLower, 1, pixelPtsOnAxes, lineModel, lineObject.key);
+      lineObject.handleUpper && fixHandles(lineObject.handleUpper, 2, pixelPtsOnAxes, lineModel, lineObject.key);
+      updateClasses(lineObject.line, lineObject.key);
+      refreshEquation(slope, intercept, lineModel, index, lineObject.key);
+    });
+  }, [annotationLocationSetter, calculateHandlePosition, instanceId, layout, model.isVisible, model.lines, plotHeight,
+      plotWidth, positionEquation, updateClasses, xAttrName, xAxis, xScale, xSubAxesCount, yAttrName, yAxis, yScale,
+      ySubAxesCount]);
+
+  // Refresh the scale copies
+  useEffect(() => {
+    const disposer = autorun(() => {
+      xScaleCopy.current = xScale.copy();
+      yScaleCopy.current = yScale.copy();
+    });
+    return () => disposer();
+  }, [xScale, yScale]);
+
+  // Refresh the lines
+  useEffect(function refresh() {
+    const disposer = autorun(() => {
+      refreshLines();
+    });
+    return () => disposer();
+  }, [refreshLines]);
 
   const
     // Line drag handler
-    continueTranslate = useCallback((event: MouseEvent) => {
-      const lineParams = model.lines?.get(instanceKey),
+    continueTranslate = useCallback((event: MouseEvent, lineKey: string) => {
+      const lineParams = model.lines.get(lineKey),
         slope = lineParams?.currentSlope || 0,
-        tWorldX = xScaleCopy.invert(event.x),
-        tWorldY = yScaleCopy.invert(event.y);
+        tWorldX = xScaleCopy.current.invert(event.x),
+        tWorldY = yScaleCopy.current.invert(event.y);
 
       // If the line is dragged outside plot area, reset it to the initial state
       if (
-          tWorldX < xScaleCopy.domain()[0] ||
-          tWorldX > xScaleCopy.domain()[1] ||
-          tWorldY < yScaleCopy.domain()[0] ||
-          tWorldY > yScaleCopy.domain()[1]
+          tWorldX < xScaleCopy.current.domain()[0] ||
+          tWorldX > xScaleCopy.current.domain()[1] ||
+          tWorldY < yScaleCopy.current.domain()[0] ||
+          tWorldY > yScaleCopy.current.domain()[1]
       ) {
         const { intercept, slope: initSlope } = computeSlopeAndIntercept(xAxis, yAxis);
-        model.dragLine(intercept, initSlope, instanceKey);
+        model.dragLine(intercept, initSlope, lineKey);
         return;
       }
 
       const newIntercept = isFinite(slope) ? tWorldY - slope * tWorldX : tWorldX;
-      model.dragLine(newIntercept, slope, instanceKey);
-    }, [instanceKey, model, xAxis, xScaleCopy, yAxis, yScaleCopy]),
+      model.dragLine(newIntercept, slope, lineKey);
+      refreshLines();
+    }, [model, refreshLines, xAxis, xScaleCopy, yAxis, yScaleCopy]),
```

Comment:
Shouldn't need to include `xScaleCopy` and `yScaleCopy` as dependents now that they're refs.